packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -51,11 +53,41 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 				canUser( 'create', 'media' ),
 				true
 			),
+			baseUrl: getUnstableBase()?.url || '',
 		};
 	}, [] );
 
 	const { undo } = useDispatch( editorStore );
 
+	// Temporary home - should this live in `core-data`?
+	function fetchRichUrlData( url, fetchOptions = {} ) {
+		const emptyDataSet = {};
+
+		const fetchingBaseUrl = '' === baseUrl;
+
+		// If the baseUrl is still resolving then return
+		// empty data for this request.
+		if ( fetchingBaseUrl ) {
+			return Promise.resolve( emptyDataSet );
+		}
+
+		// More accurate test for internal URLs to avoid edge cases
+		// such as baseURL being included as part of a query string
+		// on the target url.
+		const baseUrlAuthority = getAuthority( baseUrl );
+		const urlAuthority = getAuthority( url );
+
+		const isInternal = urlAuthority === baseUrlAuthority;
```

Comment:
Should we guard when `url` is not a valid url? Or is it being handled elsewhere?

`getAuthority` also returns undefined if there's no _authority_. TBH, maybe we should just use the native `URL` constructor to parse the urls?

Reply:
Great point. Lets follow up with guards.

Reply:
I added the `isUrl` guard because this package has been designed to  cater for URLs. I remember @aduth spending a long time on getting it to conform to the spec.

We may still need something like `isValidHref` to check whether a URL is suitable to be fetched (eg: it isn't a tel or a hash or anything like that):

https://github.com/WordPress/gutenberg/blob/eaa1be08e278a76774c040414c1691058caaa653/packages/format-library/src/link/utils.js#L29